### PR TITLE
spark: Update Dataproc run facet to include runType property

### PR DESCRIPTION
**Pull request by codelixir**
Add an additional property "runType" in Dataproc run facet to identify whether it is a job, a batch, or a session.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [x] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

**Review by ngorchakova — COMMENTED** [review]
please rename GcpDataprocSparkRunFacet to GcpDataprocRunFacet (without the word 'Spark') and the similar change should be done in all places (i.e. in json gcp_dataproc_spark -> gcp_dataproc)

File: spec/registry/gcp/dataproc/facets/GcpDataprocSparkRunFacet.json [hunk 1]
```diff
@@ -50,6 +50,10 @@
               "description": "The name of the query node in the executed Spark Plan. Often used to describe the command being executed.",
               "type": "string"
             },
+            "runType": {
```
Comment (pawel-big-lebowski) [inline spec/registry/gcp/dataproc/facets/GcpDataprocSparkRunFacet.json:53]: How can `job` be a `runType`? Perhaps I am biased by OpenLineage semantics of job and its runs. In this case `runType` equal to `job` makes little sense.

**Comment by mobuchowski** [comment]
@codelixir please rebase - this should help with few of the test failures.

File: spec/registry/gcp/dataproc/facets/GcpDataprocSparkRunFacet.json (hunk 1 above)
Comment (codelixir) [reply spec/registry/gcp/dataproc/facets/GcpDataprocSparkRunFacet.json]: I have renamed the property to `jobType`

**Comment by codelixir** [comment]
> @codelixir please rebase - this should help with few of the test failures.

I am getting this error after rebasing:
https://app.circleci.com/pipelines/github/OpenLineage/OpenLineage/13118/workflows/7f730323-8f59-4667-b71a-74f5e5d29fc9/jobs/316652

Is it related to the recent changes? Or did I do something wrong while pushing my code?

**Comment by arturowczarek** [comment]
@codelixir There was a ci/cd bug. It is fixed and merged now. Please rebase and push the changes again.

**Comment by codelixir** [comment]
> @codelixir There was a ci/cd bug. It is fixed and merged now. Please rebase and push the changes again.

Thanks, @arturowczarek!
Some of the tests are still failing but seem unrelated to my change; can one of you please re run these?

**Comment by mobuchowski** [comment]
@codelixir 

Those assertions fail:
```
GCPUtilsTest > testForDataprocCluster() FAILED
    org.opentest4j.AssertionFailedError at GCPUtilsTest.java:113

GCPUtilsTest > testForDataprocSession() FAILED
    org.opentest4j.AssertionFailedError at GCPUtilsTest.java:155

GCPUtilsTest > testForDataprocBatch() FAILED
    org.opentest4j.AssertionFailedError at GCPUtilsTest.java:134
```

**Comment by codelixir** [comment]
Thanks, @mobuchowski for pointing it out
I have fixed the unit tests

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**